### DISCUSSION TITLE
Issue when setting fzf-lua profile to `max-perf`

### DISCUSSION THREAD

**Discussion by anotherfrontendguy** [LazyVim/LazyVim#5200] [Q&A] [answered]
Hey ✌️

first of all thank you so much for all of this, you're the man folke! I really like the switch to`fzf-lua` everything feels so snappy.

There is one thing I cannot figure out, I want to set my fzf-lua [profile](https://github.com/ibhagwan/fzf-lua/tree/main/lua/fzf-lua/profiles) to `max-perf`.

When I only set the `max-perf` option, everything works like expected:
<img width="2094" alt="correct" src="https://github.com/user-attachments/assets/4c7fb382-4397-4691-8801-88ddb141b690" />

But as soon as I add any other configuration, something weird happens:
<img width="2120" alt="incorrect" src="https://github.com/user-attachments/assets/bedd1865-dd5f-47d7-ab95-024716a8d6f9" />

I can set it to any other value like `telescope` or `fzf-native` and everything works fine, just `max-perf` seems to be the odd one. I was really ambitious to fix this on my own but I'm afraid I'm at my wits end.

Anyone got an idea what's going on?

Some random key facts that might™ be related:

- `bat` & `rg` are installed.
- macOS 14.4.1
- fish shell
- NVIM v0.10.2
-  fzf 0.57.0

**Comment by dpetka2001** [comment]
Just tried a fresh LazyVim installation from scratch and couldn't reproduce. Following screencast

[Screencast_2024-12-19-20-26-21.webm](https://github.com/user-attachments/assets/c63ac41b-fed7-4ca2-b821-3400834d3c54)

I'm using Linux Mint 21.3, `bat`, `rg`, `fish` shell and also `fzf` version 0.57.0

Maybe it's something specific to Mac (I don't have so can't test)? You should open a Discussion in fzf-lua repo, so that maintainer can help you out.

**Reply by dpetka2001** [reply]
Also maybe it's a terminal thing? Could you try with Kitty or Wezterm to test if you encounter the same behavior?

**Reply by anotherfrontendguy** [reply]
Thank you for testing!

Yeah I'm also pretty sure something is messed up on my end, just don't know what it is.

The initial screenshots are of alacritty but I also tried it with kitty and (mac) builtin terminal - all with the same result (didn't try wezterm since I don't have it installed).

And yeah, good idea I'll open a discussion in the fzf-lua repo. But maybe I'll try a clean install of LazyVim first to rule out errors on my end.

**Reply by anotherfrontendguy** [reply]
ha! ok unexpected follow up: (I'll just post it here in case someone runs into the same problem)

It does indeed happen with a fresh / clean LazyVim install. I also looked into the fzf-profiles `.lua` files and tried my luck with this line: https://github.com/ibhagwan/fzf-lua/blob/main/lua/fzf-lua/profiles/max-perf.lua#L12

When I set the `--ansi` flag to true everything works as it should.  I'll try my luck in the fzf-lua discussions. Thanks again for helping out.

**Reply by dpetka2001** [reply]
Weird this does not affect Linux like in my case. Must be a MacOS or even shell thing maybe? Do you have any weird customization in fish shell to beautify it?

**Reply by anotherfrontendguy** [reply]
> Do you have any weird customization in fish shell to beautify it?

Good point but I don't think I have. I just tried changing my shell to plain `/bin/bash` and then starting the clean LazyVim install with the same strange result as before.

(For reference, this is the discussion I started in the `fzf-lua` repo: https://github.com/ibhagwan/fzf-lua/discussions/1607 )

**Comment by ibhagwan** [comment, marked as answer]
Max perf adds `--ansi`, that prevents fzf from parsing the ansi colors in the header which is a performance boost, but this is doesn’t mess up the header.

Set the options to: `{'max-perf', fzf_opts = { ['--ansi'] = true } }`, to workaround.

**Reply by dpetka2001** [reply]
Isn't this already  the default? What worked for OP is he set `--ansi = true`. I couldn't reproduce this however on Linux. Maybe a MacOS thing?

**Reply by anotherfrontendguy** [reply]
Wow you're fast - thanks!

Ok so the problem must arise elsewhere then I guess - I'll keep looking.

**Reply by ibhagwan** [reply]
> Isn't this already the default? What worked for OP is he set `--ansi = true`. I couldn't reproduce this however on Linux. Maybe a MacOS thing?

true is the default value,ANSI false is just set with max perf.

**Reply by dpetka2001** [reply]
Maybe a font thing? Or not possible?

**Reply by anotherfrontendguy** [reply]
> Set the options to: `{'max-perf', fzf_opts = { ['--ansi'] = false } }`

I'm afraid this doesn't solve it, only `['--ansi'] = true` works correctly. I don't want to cause any headache because of this, I can really just go back to setting it to `fzf-native` which works fine. I'm just curious what's happening here.

**Reply by dpetka2001** [reply]
@anotherfrontendguy Can you try with different font?

**Reply by anotherfrontendguy** [reply]
Oh sorry forgot to mention: I just tried with `Menlo` - same result.

**Reply by ibhagwan** [reply]
> > Set the options to: `{'max-perf', fzf_opts = { ['--ansi'] = false } }`
> 
> I'm afraid this doesn't solve it, only `['--ansi'] = true` works correctly. I don't want to cause any headache because of this, I can really just go back to setting it to `fzf-native` which works fine. I'm just curious what's happening here.

I meant true by bad answering from phone :)

**Reply by anotherfrontendguy** [reply]
Hehe kudos for answering coding questions via phone :) I'm totally fine with the workaround (it works) :+1:

Can you maybe change `true` to `false` in your original message? Then I'll mark it as answer (but no rush).

**Reply by ibhagwan** [reply]
> Can you maybe change true to false in your original message? Then I'll mark it as answer (but no rush).

changed as I replied to you :)
